src/histogram.c

```diff
@@ -44,45 +267,119 @@ hist_sfunc(PG_FUNCTION_ARGS)
 	double		max = DatumGetFloat8(max_datum);
 	int			nbuckets = DatumGetInt32(nbuckets_datum);
 	int32		bucket = DatumGetInt32(DirectFunctionCall4(width_bucket_float8, val_datum, min_datum, max_datum, nbuckets_datum));
+	char	   *subagg_text;
 
+
+	/* Cannot be called directly because of internal-type argument */
 	if (!AggCheckCallContext(fcinfo, &aggcontext))
-	{
-		/* cannot be called directly because of internal-type argument */
 		elog(ERROR, "hist_sfunc called in non-aggregate context");
-	}
 
+	/* Cannot generate a histogram with incompatible bounds */
 	if (min > max)
-	{
-		/* cannot generate a histogram with incompatible bounds */
 		elog(ERROR, "lower bound cannot exceed upper bound");
-	}
+
+	/* Check if bucket_agg is set, otherwise use 'count' */
+	if (PG_NARGS() == 5)
```

Comment:
Isn't it better to use `PG_ARGISNULL(5)`. Seems more explicit since the "other" case could be that `PG_NARGS()` is anything but 5.

Reply:
Done

Reply:
Actually I tried this and it kept crashing. I am not sure why.